core/selection/optimization.js

```diff
@@ -117,15 +117,17 @@
 			return false;
 		}
 
-		if ( range.endOffset === 0 ) {
+		var isInList = range.endContainer.is && range.endContainer.is( 'li' );
+
+		if ( range.endOffset === 0 && !isInList )  {
```

Comment:
Let's add a comment with a reference to the issue that this PR is solving and I think that short information about preventing shrinking selection in lists would be also nice.

```suggestion
		var isInList = range.endContainer.is && range.endContainer.is( 'li' );
        
        // Prevent shrink selection in lists (#4931).
		if ( range.endOffset === 0 && !isInList )  {
```

Also, is this additional check `range.endContainer.is` is needed? 🤔

Reply:
I'll add a comment 👍 

The additional check is for the case when container is a text node - then it doesn't have `is` method which causes errors.

Reply:
I slightly change the comment you proposed, since at this function level we don't know nth about 'shrinking' itself.